simapp/app.go

```diff
@@ -196,7 +197,7 @@ func NewSimApp(
 			depinject.Supply(
 				// supply the application options
 				appOpts,
-
+				mempool.DefaultPriorityMempool,
```

Comment:
Not sure this is relevant, but...
Above, in file `baseapp/util_test.go`, there is another call to `depinject.Supply`. In that call `DefaultPriorityMempool` is actually _called_, and its return value is provided to `depinject.Supply`, whereas here `DefaultPriorityMempool` is provided as a functor (i.e., without `()` at the end). Why this difference?

Reply:
This is relevant. It's a little confusing so I reverted to supplying a `runtime.BaseAppOption` instead in both places.  This is how we configure BaseApp throughout SDK in other places so it's more consistent.